tests/unit/test_benchmarks_examples.py

```diff
@@ -16,10 +16,19 @@
 
 # pylint: disable=import-outside-toplevel
 
+import pytest
 
-def test_decoder_example():
+
+@pytest.mark.parametrize("noise", ["cv", "dv"])
+def test_decoder_example(noise):
     """Simple test for the decoding module in flamingpy.examples."""
-    from flamingpy.examples import decoding as dc_examples
```

Comment:
What was the issue behind this approach? Is it that the codecov score becomes too low? In which case: is it preferable to create functions out of the examples and benchmarks (as you have) or simply to tell codecov to ignore the examples and benchmarks files (we do test them but only with import statements to see if they run - all the other functionality is tested in other places).

Reply:
The issue with just importing the files is that branching logic is not evaluated, lowering the file coverage. Out of the two possible solutions, _create functions out of the examples and benchmarks_ or _simply to tell codecov to ignore the examples and benchmarks files_, I chose the former as it will always ensure that the examples are in working state when updates to the codebase are applied. However, just removing the files from coverage will also do.

What do you think?